lib/foreman_maintain/concerns/system_helpers.rb

```diff
@@ -188,6 +193,19 @@ def find_dir_containing_file(directory, target)
         end
         result
       end
+
+      private
+
+      def check_version(name)
+        current_version = package_version(name)
+        if current_version
+          yield current_version if block_given?
```

Comment:
Since there is no reason to call `check_version` without a block, I would be ok not checking on presence of the block here.